File: solutions/javascript/acronym/1/acronym.js
let delimiter = /[\s-]+/;
let firstChar = /[^a-zA-Z]*([a-zA-Z])/;

export const parse = (phrase) => {
  return phrase
    .split(delimiter)
    .map((word) => {
      const matches = word.match(firstChar);
      return matches != null || matches.length > 1 ? matches[1] : "";
    })
    .join("")
    .toUpperCase();
};

File: solutions/java/acronym/1/src/main/java/Acronym.java
import java.util.Arrays;
import java.util.stream.Collectors;

class Acronym {
    String phrase;
    String acronym;

    Acronym(String phrase) {
        this.phrase = phrase;

        this.acronym = String.join("",
                Arrays.asList(phrase.split("[- ]")).stream().map(str -> str.replaceAll("[^a-zA-Z]", ""))
                        .filter(str -> !str.isEmpty()).map(str -> str.substring(0, 1).toUpperCase())
                        .collect(Collectors.toList()));
    }

    String get() {
        return this.acronym;
    }

}
